clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/CloudMetricController.groovy

```diff
@@ -0,0 +1,102 @@
+import com.netflix.spinnaker.clouddriver.model.CloudMetricProvider
+import com.netflix.spinnaker.clouddriver.model.CloudMetricStatistics
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.time.Clock
+
+@RequestMapping("/cloudMetrics")
+@RestController
+class CloudMetricController {
+
+  @Autowired
+  final List<CloudMetricProvider> metricProviders
+
+  final Clock clock
+
+  @RequestMapping(method = RequestMethod.GET, value = "/{cloudProvider}/{account}/{region}")
+  List<CloudMetricDescriptor> findAll(@PathVariable String cloudProvider,
+                        @PathVariable String account,
+                        @PathVariable String region,
+                        @RequestParam Map<String, String> filters) {
+
+    getProvider(cloudProvider).listMetricDescriptors(account, region, filters)
+  }
+
+  CloudMetricController(List<CloudMetricProvider> metricProviders, Clock clock) {
+    this.clock = clock;
+    this.metricProviders = metricProviders
+  }
+
+  @Autowired
+  CloudMetricController(List<CloudMetricProvider> metricProviders) {
+    this(metricProviders, Clock.systemDefaultZone())
+  }
+
+  /**
+   * Returns a data set describing the metric over the specified period of time. If start and end times are not provided,
+   * they will be defaulted to 24 hours ago and now, respectively
+   * @param cloudProvider the provider, e.g. "aws", "gce", etc.
+   * @param account the account of the provider
+   * @param region the region applicable to the metric
+   * @param metricName the name of the metric
+   * @param startTime start time (inclusive)
+   * @param endTime end time (exclusive)
+   * @param filters any provider-specific filters used to create the data set
+   * @return the set of statistics
+   */
+  @RequestMapping(method = RequestMethod.GET, value = "/{cloudProvider}/{account}/{region}/{metricName}/statistics")
+  CloudMetricStatistics getStatistics(@PathVariable String cloudProvider,
+                                      @PathVariable String account,
+                                      @PathVariable String region,
+                                      @PathVariable String metricName,
+                                      @RequestParam(required = false) Long startTime,
+                                      @RequestParam(required = false) Long endTime,
+                                      @RequestParam Map<String, String> filters) {
+
+    // Spring will map all request parameters into the filters map, so it's best to remove the declared ones
+    filters.remove("startTime")
+    filters.remove("endTime")
+
+    Long start = startTime ?: clock.instant().minus(24, HOURS).toEpochMilli()
```

Comment:
Maybe this could be a configurable parameter so you can specify the window you want to use here. A day seems pretty long for autoscaling decisions. I would have guessed you'd be using a much smaller window.

On the end time, I wonder if you want it to be open ended to be the most current data you have. This will wind up with a slightly bigger window but means in practice you wont need to strip off data (unless the endTime is projected into the future to achieve the same effect). Otherwise when you say "now" you are looking into the past (by the time the request is processed) rather than the present. Leaving it open ended lets you defer the end time until you are forced (if at all) to apply an actual value.